### PR TITLE
(#121) Allow top-level choria::server_config options to be arrays

### PR DESCRIPTION
**Pull request by snakeb1t**
I think this will work, but I need a little help with writing a test for it

### CODE REVIEW

**Comment by ripienaar** [comment]
hmm, this have come up a few times but I am trying to think why I never accepted it, there's some common case that doesnt work - I'll think a bit :) 

happy to help with the test a bit later

**Comment by snakeb1t** [comment]
I think we need to use https://github.com/mthibaut/hiera-puppet-helper ? But when I try to install it locally, I get a bunch of errors when running tests, even if I don't try to add new ones.

**Comment by ripienaar** [comment]
For data testing? Nah no need for hiera there just Pass the data you want to test as class pranksters when testing. 

We can assume hiera functions as it should - not our concern. So I think other tests just Pass the data.

File: functions/hash2config.pp [hunk 1]
```diff
@@ -2,7 +2,11 @@
 # key = value
 function choria::hash2config(Hash $confighash) >> String {
   $result = $confighash.keys.sort.map |$key| {
-    sprintf("%s = %s", $key, $confighash[$key])
+    $val = type($confighash[$key], 'generalized') ? {
+      /Array/ => ($confighash[$key] << []).join(", "),
+      default => $confighash[$key],
+    }
+    sprintf("%s = %s", $key, $val)
```
Comment (ripienaar) [inline functions/hash2config.pp]: give this a go

```
$result = $confighash.keys.sort.map |$key| {
    $val = $confighash[$key]

    if $val =~ Array {
      $val.each |$item| {
        unless $item =~ Integer or $item =~ Float or $item =~ String {
          fail("Array values are only supported for Intger, Float or String data types")
        }
      }

      sprintf("%s = %s", $key, $val.join(", "))
    } else {
      sprintf("%s = %s", $key, $val)
    }
}
```

Comment (snakeb1t) [reply functions/hash2config.pp]: Heyyyy, that worked. You the man!

**Pull request merged**